sdk/search/Azure.AI.Search/src/SearchClientOptions.cs

```diff
@@ -0,0 +1,173 @@
+        /// client library.
+        /// </exception>
+        public SearchClientOptions(ServiceVersion version = LatestVersion)
+        {
+            Version = version.Validate();
+            AddLoggingHeaders();
+            AddLoggingQueryParameters();
+        }
+
+        /// <summary>
+        /// Create an <see cref="HttpPipeline"/> to send requests to the Search
+        /// Service.
+        /// </summary>
+        /// <param name="credential">
+        /// The <see cref="SearchApiKeyCredential"/> to authenticate requests.
+        /// </param>
+        /// <returns>An <see cref="HttpPipeline"/> to send requests.</returns>
+        internal HttpPipeline Build(SearchApiKeyCredential credential)
+        {
+            Debug.Assert(credential != null);
+            return HttpPipelineBuilder.Build(
+                options: this,
+                perCallPolicies: new[] { new SearchApiKeyCredentialPolicy(credential) },
+                perRetryPolicies: Array.Empty<HttpPipelinePolicy>(),
+                responseClassifier: null);
+        }
+
+        /// <summary>
+        /// Add the allow list headers to the <see cref="DiagnosticsOptions"/>
+        /// that are considered safe for logging/exceptions by default.
+        /// </summary>
+        private void AddLoggingHeaders()
+        {
+            Diagnostics.LoggedHeaderNames.Add("x-ms-version");
+            Diagnostics.LoggedHeaderNames.Add("OData-version");
+            Diagnostics.LoggedHeaderNames.Add("elapsed-time");
+            // TODO: Add the rest...
+        }
+
+        /// <summary>
+        /// Add the allow list query parameters to the
+        /// <see cref="DiagnosticsOptions"/> that  ar considered safe for
+        /// logging/exceptions by default.
+        /// </summary>
+        private void AddLoggingQueryParameters()
+        {
+            Diagnostics.LoggedQueryParameters.Add("api-version");
+            // TODO: Add the rest...
+        }
+    }
+
+    /// <summary>
+    /// Search extension methods.
+    /// </summary>
+    internal static partial class SearchExtensions
+    {
+        /// <summary>
+        /// Validate a <see cref="SearchClientOptions.ServiceVersion"/>.
+        /// </summary>
+        /// <param name="version">
+        /// The <see cref="SearchClientOptions.ServiceVersion"/> to validate.
+        /// </param>
+        /// <returns>
+        /// The validated version.
+        /// </returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when the <paramref name="version"/> is not supported by this
+        /// client library.
+        /// </exception>
+        public static SearchClientOptions.ServiceVersion Validate(this SearchClientOptions.ServiceVersion version) =>
+            version switch
+            {
+                SearchClientOptions.ServiceVersion.V2019_05_06 => version,
+                _ => throw CreateInvalidVersionException(version)
```

Comment:
Just a thought, but in KV we throw `NotSupportedException`.

Reply:
The [guidelines throw an ArgumentException](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-versioning).